libs/did-resolver-lib/src/resolver.ts

```diff
@@ -0,0 +1,125 @@
+import { DIDDoc, DIDResolver, Service, VerificationMethod } from 'didcomm';
+import base64url from 'base64url';
+
+type Purpose =
+  | 'Assertion'
+  | 'Encryption'
+  | 'Verification'
+  | 'CapabilityDelegation'
+  | 'CapabilityInvocation'
+  | 'Service';
+
+export default class PeerDIDResolver implements DIDResolver {
+  async resolve(did: string): Promise<DIDDoc | null> {
+    try {
+      // Validate if the DID starts with the "did:peer:" prefix
+      if (!did.startsWith('did:peer:')) {
+        throw new Error('Unsupported DID method');
+      }
+
+      // Dissect the DID address
+      const chain = did
+        .replace(/^did:peer:2\./, '')
```

Comment:
You only seem to handle did:peer:2, which is okay. But you should throw a specific error when the function is given another kind of did:peer address.

Reply:
Ops i though i handled this